### PR TITLE
Update promoteLambda

### PR DESCRIPTION
**Pull request by cwgoes**
Fetch the type of closures.

### CODE REVIEW

File: src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs [hunk 1]
```diff
@@ -26,6 +26,7 @@ import qualified Juvix.Core.Usage as Usage
 import Juvix.Library hiding (abs, and, or, xor)
 import qualified Juvix.Library (abs)
 import qualified Michelson.Untyped.Instr as Instr
+import qualified Michelson.Untyped.Type as MU
```
Comment (mariari) [inline src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs]: We should be able to do this without importing this. 

```haskell
Juvix.Backends.Michelson.DSL.Untyped
```
should be used instead

File: src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs [hunk 2]
```diff
@@ -34,14 +35,16 @@ import Prelude (error)
...
 
 instOuter ∷ Env.Reduction m ⇒ Types.NewTerm → m Instr.ExpandedOp
-instOuter a@(Types.Ann _ ty _) = inst a >>= expandedToInst ty
+instOuter a@(Types.Ann _ ty _) = do
+  inst ← inst a
+  ty ← typeToPrimType ty
+  expandedToInst ty inst
 
-expandedToInst ∷ Env.Reduction m ⇒ Types.Type → Env.Expanded → m Instr.ExpandedOp
+expandedToInst ∷ Env.Reduction m ⇒ MU.Type → Env.Expanded → m Instr.ExpandedOp
```
Comment (mariari) [inline src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs]:
```suggestion
expandedToInst ∷ Env.Reduction m ⇒ Untyped.T → Env.Expanded → m Instr.ExpandedOp
```

File: src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs [hunk 3]
```diff
@@ -619,18 +622,35 @@ eatType _ _ = error "Only eat parts of a Pi types, not any other type!"
 -- Misc
 --------------------------------------------------------------------------------
 
+mustLookupType ∷
+  ( HasState "stack" (VStack.T lamType) m,
+    HasThrow "compilationError" Types.CompilationError m
+  ) ⇒
+  Symbol →
+  m MU.Type
```
Comment (mariari) [inline src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs]:
```suggestion
  m Untyped.T
```

File: src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs (hunk 1 above)
Comment (cwgoes) [reply src/Juvix/Backends/Michelson/DSL/InstructionsEff.hs]: Right, fixed.

**Pull request merged**